src/rust/engine/rule_graph/src/builder.rs

```diff
@@ -213,9 +224,7 @@ impl<R: Rule> Builder<R> {
 
   pub fn graph(self) -> Result<RuleGraph<R>, String> {
     // 1. build a polymorphic graph
-    //    * only consideration: whether something is a valid input Param _anywhere_ in the graph, and
-    //      that we can't satisfy a Get with a Param
```

Comment:
This type of prior explanation was really helpful. Is there a meaningful short sentence that you could add that explains what types of nodes will be in this "polymorphic graph"? A tl;dr of what analysis is being done in this stage.